qualify quicklz references

lisakowen opened the pull request:
trying out the gpdb docs PR process with the changes for qualifying quicklz.  also my first edits with oxygen. (i am not really sure how to interpret the diffs and tell if my editting is ok.)  local build looked ok.  i don't expect this PR to be accepted as is.

anyway, i approached the changes in a couple of different ways, the first time quicklz was referenced on the page outside of reference page syntax definition:
- added a numbered superscript and note
- added just a note
- ALTER TABLE had one reference to quicklz in a syntax definition and no further discussion.  this page may be missing a section?  in any case, i didn't do anything on this page because because no discussion in which to add the qualifier.

also, wasn't sure if i needed to qualify every time quicklz was used.  input welcome!  thanks.

pivotal-issuemaster commented:
@lisakowen Please sign the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=1686)!

[Click here](https://cla.pivotal.io/sync/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=1686) to manually synchronize the status of this Pull Request.

See the [FAQ](https://cla.pivotal.io/about) for frequently asked questions.

pivotal-issuemaster commented:
@lisakowen Thank you for signing the [Contributor License Agreement](https://cla.pivotal.io/sign/greenplum?repositoryId=greenplum-db/gpdb&pullRequestId=1686)!

mkiyama commented:
I believe ALTER TABLE points back to CREATE TABLE for a lot of stuff.

Do you want to say "supported" or "provided"?

I believe if you specify QUICKLZ as a compression type in OSS GPDB, an error is returned.
I think somebody on the catalog and storage team might know.

Eulerizeit commented:
Quicklz compression is not part of the OSS build.

lisakowen commented:
re: oss & quick lz, the docs currently say (on http://docs-greenplum-db-staging.cfapps.io/docs-gpdb/common/pivotal-gpdb-features.html)

The QuickLZ compression method is not supported in Greenplum Database and is ignored if specified.

asubramanya commented:
The OSS GPDB is not built with quicklz support.

Seems like creation of an AO table with `quicklz` goes through:
```
postgres=# create table ao1(a int, b int) with(appendonly=true, compresstype='quicklz');
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'a' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
CREATE TABLE
```

However Trying to `select / insert` into an AO table with `quicklz` compression support errors out as follows:
```
postgres=# insert into ao1 values(1, 1);
ERROR:  quicklz compression not supported (quicklz_compression.c:26)  (seg0 127.0.0.1:25432 pid=18933)
postgres=# select * from ao1;
ERROR:  quicklz compression not supported (quicklz_compression.c:26)  (seg0 slice1 127.0.0.1:25432 pid=18933)
```

danielgustafsson commented:
The quicklz support is not added as a traditional module in OSS GPDB, the proprietary part is instead replaced with the stubs returning the above error. Ideally we should make quicklz a configure time parameter like `./configure --with-quicklz=<PATH>` or something, not sure how much trouble it's worth though.

mkiyama commented:
Just for completeness, and stating the obvious...
This behavior should be documented, either as a known limitation or as intended functionality, depending on what dev decides.

lisakowen commented:
documenting the implementation as it currently works.  verified that create table operations with quicklz specified for table-, column-, and partition-level compression type succeed with no error or warning.  subsequent insert and select operations on these tables fail with the "quicklz not supported" error.

updates include:
- added a limitation to gpdb 5.0.0 release notes about this feature
- clarified the quicklz statement in the "about this documentation" section that summarizes differences between open source and pivotal gpdb
- added a note at first occurrence of quicklz use in create table, create table as, pg_appendonly catalog, choosing the table storage model pages

TODO:
- update GUC page when docs build is fixed.

mkiyama commented on gpdb-doc/dita/relnotes/GPDB_500_README.xml:
```diff
@@ -116,6 +110,22 @@
+      <ul id="ul_lg5_pyd_wy">
+        <li>
+          <p>GPDB 5.0.0 does not provide QuickLZ compression, but accepts <codeph>CREATE TABLE
+              [AS]</codeph> operations with table-, partition-, and/or column-level
```
Typo- extra "s" after compresstype
I would put the ERROR in <codeph>
